### PR TITLE
Unsupported theme enhancements

### LINKED ISSUE #13087: Improve "unsupported" theme experience
To improve basic appearance when using a non-wc compatible theme (this can be detected) we should offer a cut-down, content area only view for store pages. This avoids the need to work with [wrappers](https://github.com/woocommerce/woocommerce/issues/17549).

We could use shortcodes, but I think we could also filter on the_content and the_title of the shop page to do what we need without [changing page content to a shortcode as suggested here](https://github.com/woocommerce/woocommerce/issues/13087#issuecomment-308159717).
Either solution may work so some experimentation might be in order.

If theme supports WooCommerce, do nothing different. Use the current template system, and archives etc.

<details>

This was the result of a discussion with Jay about how we can make the 'default' WooCommerce theme experience better, if not using a WooTheme such as Storefront.

## The problem

Currently, if you're using a third party theme, due to the way templates work in WooCommerce, pages like the shop page will have odd layouts until these steps are performed:

https://docs.woocommerce.com/document/third-party-custom-theme-compatibility/

The options being:

- Adding correct 'wrappers' so our templates know how to format the headers and footers
- Using the `woocommerce_content()` function in a page.
- Theming it fully.

The main reasons for these are:

- WooCommerce uses custom post type archives, single post type, and real archive templates for the shop/category archives/single pages. These require a template to work in most cases.
- Shop and product content is not something we'd usually format like a post, and the content of a product for example can span the whole page. Same for the archives; we'd usually show sidebars and widgets related to the shop content.

Currently, the main non-pages WooCommerce generates are:

- Category, tag, and attribute archives.
- The product post archive (shop page).
- Single product pages.

## Potential solution

We'd like to experiment with a shortcode based approuch so that the theme's page.php file will be responsible for shop content, and the wrappers will work out of the box.

The simplest idea being we:

- Generate pages for the shop content
- Add a shortcode, possibly a wrapper for `woocommerce_content()` (?) and add this to the pages.
- Make sure conditonal tags and permalinks still work as best as we can.
- Filter 'post titles' so we can give the page a correct title, similar to how we do the account page right now.

*Pros:*

- Better experience out of the box
- No loss of theming ability should they want something better
- Less theming tickets

*Cons:*

- The pages won't be a 'full featured'
- The shop content will need to fit with the 'page' content
- Adjusting titles on the fly can be a little 'hacky'

---

This is semi-related to #13088 where we want to push Storefront to users during setup for the best possible experience if setting up a dedicated store.

</details>

### PR DESCRIPTION
**Pull request by claudiulodro**
Closes #13087

This PR makes WooCommerce look good and work well on themes that don't formally support WooCommerce.

To test: Try WooCommerce using random themes that don't formally support WooCommerce. Also try this PR using a theme that supports WooCommerce to make sure we didn't break the default behavior.

### CODE REVIEW

**Comment by claudiulodro** [comment]
@mikejolley figured out how we can easily do product archives on unsupported themes. If it's an archive, grab the term ID from the queried object then redirect to the shop page with the category URL param set to filter on the category.

**Comment by mikejolley** [comment]
@claudiulodro let's chat later. I think it may cause issues with SEO and so on.

File: includes/shortcodes/class-wc-shortcode-products.php [hunk 1]
```diff
@@ -451,35 +488,42 @@ protected function get_transient_name() {
...
 	 */
-	protected function get_products_ids() {
+	protected function get_query_results() {
 		$transient_name = $this->get_transient_name();
-		$ids            = get_transient( $transient_name );
+		$cache          = wc_string_to_bool( $this->attributes['cache'] ) === true;
+		$data           = $cache ? get_transient( $transient_name ) : false;
 
-		if ( false === $ids ) {
+		if ( false === $data ) {
```
Comment (claudiulodro) [inline includes/shortcodes/class-wc-shortcode-products.php]: @mikejolley alternating between `$data` and `$results` here.

**Comment by tiagonoronha** [comment]
After a conversation with @mikejolley about the new `min` and `max` column settings, we concluded that there's no way for a theme to set a default number of columns.

Yes, it's possible to manually set the `woocommerce_catalog_columns` option, but I'd prefer not to have themes do that. Since WooCommerce is already handling the `min` and `max`, it would be great to have a way to set the default number.

I suggest adding a new `default_columns` parameter.

The same applies to the number of rows.

**Pull request merged**